### PR TITLE
Add complexity limitation for GraphQL query to avoid malicious query

### LINKED ISSUE #8694: [Feature] Support limitation and circuit break per GraphQL request
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

_No response_

### Use case

By leveraging the GraphQL query, users could initial a 1k+ sub GraphQL queries/mutations in one single HTTP request. Then this could crash the OAP server.

It is better we could provide a feature with a configuration to set up the number limitation. And response error if too many requests in one HTTP call.

We need to check whether we could do this in today's GraphQL framework.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kezhenxu94**
- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #8694 .
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by kezhenxu94** [comment]
Drafting to see whether the default value is suitable for all tests

File: docs/en/setup/backend/configuration-vocabulary.md [hunk 1]
```diff
@@ -230,6 +230,7 @@ core|default|role|Option values: `Mixed/Receiver/Aggregator`. **Receiver** mode
 | - | - | sampleRate | Sampling rate for receiving trace. Precise to 1/10000. 10000 means sampling rate of 100% by default. | SW_RECEIVER_BROWSER_SAMPLE_RATE | 10000 |
 | query | graphql | - | GraphQL query implementation. | - |
 | - | - | enableLogTestTool | Enable the log testing API to test the LAL. **NOTE**: This API evaluates untrusted code on the OAP server. A malicious script can do significant damage (steal keys and secrets, remove files and directories, install malware, etc). As such, please enable this API only when you completely trust your users. | SW_QUERY_GRAPHQL_ENABLE_LOG_TEST_TOOL | false |
+| - | - | maxQueryComplexity | Maximum complexity allowed for the GraphQL query that can be used to abort a query if the total number of data fields queried exceeds the defined threshold. | SW_QUERY_MAX_QUERY_COMPLEXITY | 30 |
```
Comment (wu-sheng) [inline docs/en/setup/backend/configuration-vocabulary.md:233]: Let's make it 50 or 100. Today new dashboard could set up a complex page.

File: .github/workflows/skywalking.yaml [hunk 2]
```diff
@@ -18,17 +18,18 @@ name: CI
...
-      - "*.txt"
-      - "skywalking-ui"
-      - ".asf.yaml"
-      - ".dlc.yaml"
-      - ".licenserc.yaml"
-      - "docs/menu.yml"
-      - ".github/workflows/codeql.yaml"
-      - "dist-material/release-docs"
-      - "*/component-libraries.yml"
+    paths:
```
Comment (wu-sheng) [inline .github/workflows/skywalking.yaml:21]: What does this `paths-ignore` to `path` mean? The following patterns should be CI irrelative?

Comment (kezhenxu94) [reply .github/workflows/skywalking.yaml]: There are some issues in the conditions to run / not to run the workflows, will take another look, just reverted these

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**